Explorer/Assets/Scripts/CrdtEcsBridge/JsModulesImplementation/SDKObservableEventsEngineAPIImplementation.cs

```diff
@@ -0,0 +1,44 @@
+using CRDT.Deserializer;
+using CRDT.Protocol;
+using CRDT.Protocol.Factory;
+using CRDT.Serializer;
+using CrdtEcsBridge.OutgoingMessages;
+using CrdtEcsBridge.PoolsProviders;
+using CrdtEcsBridge.UpdateGate;
+using CrdtEcsBridge.WorldSynchronizer;
+using DCL.ECS7;
+using SceneRunner.Scene.ExceptionsHandling;
+using SceneRuntime.Apis.Modules.EngineApi.SDKObservableEvents;
+using System.Collections.Generic;
+using Utility.Multithreading;
+
+namespace CrdtEcsBridge.JsModulesImplementation
+{
+    public class SDKObservableEventsEngineAPIImplementation : EngineAPIImplementation, ISDKObservableEventsEngineApi
+    {
+        public HashSet<ProcessedCRDTMessage> OutgoingCRDTMessages { get; } = new ();
+
+        public SDKObservableEventsEngineAPIImplementation(ISharedPoolsProvider poolsProvider, IInstancePoolsProvider instancePoolsProvider, ICRDTProtocol crdtProtocol, ICRDTDeserializer crdtDeserializer, ICRDTSerializer crdtSerializer, ICRDTWorldSynchronizer crdtWorldSynchronizer, IOutgoingCRDTMessagesProvider outgoingCrtdMessagesProvider, ISystemGroupsUpdateGate systemGroupsUpdateGate, ISceneExceptionsHandler exceptionsHandler, MutexSync mutexSync) : base(poolsProvider, instancePoolsProvider, crdtProtocol, crdtDeserializer, crdtSerializer, crdtWorldSynchronizer, outgoingCrtdMessagesProvider, systemGroupsUpdateGate, exceptionsHandler, mutexSync)
+        {
+        }
+
+        protected override void SyncCRDTMessage(ProcessedCRDTMessage message)
+        {
+            OutgoingCRDTMessages.Add(message);
```

Comment:
It's related to my previous review: you can't just add the message to an "external" collection because its lifecycle is not under control of this class, and thus you can stumble upon the situation when its data (bytes) is already disposed of.
I propose you to make a copy instead, store it, and then in `Clear` handle its proper disposal.